core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3864,6 +4353,37 @@ private int validateColumnAndGetModelIndex(QueryModel model, CharSequence column
         return index;
     }
 
+    /* Throws exception if given node tree contains reference to aggregate or window function that are not allowed in GROUP BY clause. */
+    private void validateGroupByExpression(@Transient ExpressionNode node, QueryModel model, int originalNodePosition) throws SqlException {
```

Comment:
At the moment it's not possible to put a aggregate or window function as a function argument because of : 
- "Aggregate function cannot be passed as an argument" error or
- parse errors - putting row_number() within a function fails with a number of messages .